components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3194,11 +2903,662 @@ subroutine rain_immersion_freezing(t,    &
+   qinuc,ninuc)
+
+   !................................................................
+   ! deposition/condensation-freezing nucleation
+   ! allow ice nucleation if < -15 C and > 5% ice supersaturation
+   ! use CELL-AVERAGE values, freezing of vapor
+
+   implicit none
+   
+   real(rtype), intent(in) :: t
+   real(rtype), intent(in) :: inv_rho
+   real(rtype), intent(in) :: nitot 
+   real(rtype), intent(in) :: naai 
+   real(rtype), intent(in) :: supi 
+   real(rtype), intent(in) :: odt 
+   logical, intent(in) :: log_predictNc 
+
+   real(rtype), intent(inout) :: qinuc 
+   real(rtype), intent(inout) :: ninuc 
+
+
+   real(rtype) :: dum, N_nuc, Q_nuc 
+   if ( t .lt.icenuct .and. supi.ge.0.05_rtype) then
+      if(.not. log_predictNc) then 
+!         ! dum = exp(-0.639+0.1296*100.*supi(i,k))*1000.*inv_rho(i,k)  !Meyers et al. (1992)
+         dum = 0.005_rtype*exp(0.304_rtype*(zerodegc-t))*1000._rtype*inv_rho   !Cooper (1986)
+         dum = min(dum,100.e3_rtype*inv_rho)
+         N_nuc = max(0._rtype,(dum-nitot)*odt)
+         if (N_nuc.ge.1.e-20_rtype) then
+            Q_nuc = max(0._rtype,(dum-nitot)*mi0*odt)
+            qinuc = Q_nuc
+            ninuc = N_nuc 
+         endif
+      else 
+      ! Ice nucleation predicted by aerosol scheme 
+         ninuc = max(0._rtype, (naai - nitot)*odt) 
+         qinuc = ninuc * mi0 
+      endif 
+   endif 
+
+end subroutine 
+
+
+subroutine droplet_activation(t,pres,qv,qc,inv_rho,sup,xxlv,npccn,log_predictNc,odt,it,    &
+   qcnuc,ncnuc)
+
+
+implicit none 
+
+real(rtype), intent(in) :: t
+real(rtype), intent(in) :: pres
+real(rtype), intent(in) :: qv  
+real(rtype), intent(in) :: qc
+real(rtype), intent(in) :: inv_rho 
+real(rtype), intent(in) :: sup 
+real(rtype), intent(in) :: xxlv 
+real(rtype), intent(in) :: npccn 
+
+logical, intent(in) :: log_predictNc
+real(rtype), intent(in)  :: odt 
+integer, intent(in) :: it 
+
+real(rtype), intent(inout) :: qcnuc 
+real(rtype), intent(inout) :: ncnuc 
+
+real(rtype) :: dum, dumqvs, dqsdt, ab 
+
+!.................................................................
+! droplet activation
+
+   if (log_predictNc) then
+      ! for predicted Nc, use activation predicted by aerosol scheme
+      ! note that this is also applied at the first time step
+      if (sup.gt.1.e-6) then
+         ncnuc = npccn
+         if (it.eq.1) then
+            qcnuc = 0._rtype
+         else
+            !TODO Limit qcnuc so that conditions never become sub-saturated 
+            qcnuc = ncnuc*cons7
+         endif
+      endif
+   else if (sup.gt.1.e-6.and.it.gt.1) then
+     ! for specified Nc, make sure droplets are present if conditions are supersaturated
+     ! this is not applied at the first time step, since saturation adjustment is applied at the first step
+      dum   = nccnst*inv_rho*cons7-qc
+      dum   = max(0._rtype,dum)
+      dumqvs = qv_sat(t,pres,0)
+      dqsdt = xxlv*dumqvs/(rv*t*t)
+      ab    = 1._rtype + dqsdt*xxlv*inv_cp
```

Comment:
I don't believe that T changes from when the `get_time_space_phys_variables` subroutine is called and `droplet_activation` is called.  Could we just pass the variables `ab` and `dumqvs` directly in and use the values available?  They would be `ab` and `qvs(i,k)` in the main code.

Reply:
A follow up on this, I tested this theory with an ne30 run going out to 3 steps and found it was BFB if we use the `ab` and `qvs` passed directly in instead of calculating them in the subroutine.

Reply:
As with my other recent comment, this only kicks in when log_predictNc is .false. which isn't the default.  So it is no surprise my runs were BFB since I wasn't testing this case anyways.  Probably best to just leave this in as is for now and we will be revisiting this chunk of code entirely in another PR.

Reply:
This code will be removed when we resolve https://github.com/E3SM-Project/scream/issues/121#issue-474694720 .